r-package/R/bdplyr.R

```diff
@@ -0,0 +1,532 @@
+#'
+#' @examples
+#'
+#' \dontrun{
+#'
+#' # set project billing id
+#' basedosdados::set_billing_id("avalidprojectbillingid")
+#'
+#' # connects to the remote table I want
+#' base_sim <- bdplyr("br_ms_sim.municipio_causa_idade")
+#'
+#' # connects to another remote table
+#' municipios <- bdplyr("br_bd_diretorios_brasil.municipio")
+#'
+#' # explore data
+#' base_sim %>%
+#'   dplyr::glimpse()
+#'
+#' # use normal `{dplyr}` operations
+#' municipios %>%
+#'   dplyr::select(dplyr::everything()) %>%
+#'   head()
+#'
+#' # filter
+#' base_sim_acre <- base_sim %>%
+#'  dplyr::mutate(ano = as.numeric(ano)) %>%
+#'   dplyr::filter(sigla_uf == "AC", ano >= 2018)
+#'
+#' municipios_acre <- municipios %>%
+#'   dplyr::filter(sigla_uf == "AC") %>%
+#'   dplyr::select(id_municipio, municipio, regiao)
+#'
+#'
+#' # join
+#' base_junta <- base_sim_acre %>%
+#'   dplyr::left_join(municipios_acre,
+#'                    by = "id_municipio")
+#'
+#' # tests whether the result is satisfactory
+#' base_junta
+#'
+#' # collect the result
+#' base_final <- base_junta %>%
+#'   basedosdados::bd_collect()
+#'
+#' # alternatively, write in disk the result
+#'
+#' base_final %>%
+#'   basedosdados::bd_write_rds(path = "data-raw/data.rds")
+#'
+#'}
+
+#TODO: usar bigrquery::bq_has_token() para checar se atenticou no Google
+#TODO: tentar autenticar em silêncio com bigrquery::bq_auth(email = <email>)
+# conferir: https://github.com/r-dbi/bigrquery/blob/main/R/bq-auth.R
+#TODO: usar uma vignette pra explicar a compatibilidade
+
+# bdplyr ------------------------------------------------------------------
+
+bdplyr <- function(
+  table,
+  billing_project_id = basedosdados::get_billing_id()) {
+
+  # checa se o billing id foi informado
+
+  if(billing_project_id == FALSE) {
+
+    rlang::abort("You haven't set a Project Billing Id. Use the function `set_billing_id` to do so.")
+
+  }
+
+  # criar o nome que o BQ vai entender
+  #TODO: avaliar se colocamos o parametro project livre
+
+    if (stringr::str_detect(table, pattern = "^basedosdados\\.") == TRUE) {
+
+    tabela_full_name <- table
+
+  } else {
+
+    tabela_full_name <- glue::glue("basedosdados.{table}")
+  }
+
+
+  # checa se a tabela é reconhecida pelo google
+
+  if(bigrquery::bq_table_exists(tabela_full_name) == FALSE) {
+
+    rlang::abort(glue::glue("The table {tabela_full_name} doesn´t have a valid name or was not found at basedosdados."))
+
+  }
+
+  # cria a conexão
+
+    con <- DBI::dbConnect(
+    drv = bigrquery::bigquery(),
+    project = "basedosdados",
+    billing = billing_project_id)
+
+  # chama o dplyr e guarda em um objeto
+  tibble_connection <- dplyr::tbl(con, tabela_full_name)
+
+
+  # testa se funcionou
+  if (is_tbl_lazy(tibble_connection) == TRUE) {
+    rlang::inform(glue::glue("The table `{tabela_full_name}` was successfully connected."))
+    return (tibble_connection)
+
+  } else {
+    rlang::abort(glue::glue("Error when trying to connect the table `{tabela_full_name}`"))
+  }
```

Comment:
Melhor mudar para "Successfully connected to table `{tabela_full_name}`" e reescrever esse "when trying to connect", ficou meio esquito hehe.

Reply:
melhor deixar o nome de todas as variáveis em ingles, `table_full_name` seria mais apropriado.

Acho que os comentários internos também deveriam ser em ingles pra manter um padrão